variables.tf

```diff
@@ -92,3 +92,27 @@ variable "drg_display_name" {
   type        = string
   default     = null
 }
+
+# routing rules
+
+variable "internet_gateway_route_rules" {
```

Comment:
Try changing the type to map(any) instead. The format for users to pass these rules will be difficult. 

See if example in oke module: https://github.com/oracle-terraform-modules/terraform-oci-oke/blob/master/variables.tf#L295 and this is how we use it: https://github.com/oracle-terraform-modules/terraform-oci-oke/blob/master/modules/oke/nodepools.tf#L4

The format for users to pass this will be much simpler then

Reply:
`destination`, `destination_type` and `network_entity_id` are required by API. Only `description` is optional.
So I think having a descriptive schema is less guesswork for the user and enforce what the api expect.

Reply:
By all means, keep the descriptive schema. I think you'll need it anyway. But use a map of maps if you can. It makes it easier when writing the rules.

Reply:
I agree about map(any) on other places, but for route_rules (and security_rules), I think list(object) is more adapted.

I started with map(object) then changed for list(object) because rules don't really have a one word name that could be used as a key.

You will have to make one that is not optimal (description field is better usage to give context), and when you have a route table with (lot) of rules, naming each of them is a major overhead instead of just dropping a new element object.